Fix `torch.backends.cudnn` mypy error

ShawnZhong opened the pull request:
Fix #38410

![image](https://user-images.githubusercontent.com/6421097/82724121-74b26880-9c99-11ea-9b63-e92de2dccdf2.png)

ezyang commented on torch/backends/cudnn/__init__.pyi:
```diff
@@ -0,0 +1,20 @@
+
+import torch
+from torch.backends import PropModule
+from torch.types import Number, Tuple, _bool, _dtype
```
Put these annotations inline in `cudnn/__init__.py` please! Thanks!

ezyang reviewed (changes requested):
Thank you, this is much appreciated. If it's not too much to ask, could you move the pyi annotations inline into the py file? It will be much easier to maintain that way. Thanks!

ezyang commented on torch/_C/_cudnn.pyi:
```diff
@@ -0,0 +1,17 @@
+# Defined in torch/csrc/cuda/shared/cudnn.cpp
+is_cuda: _bool
+
+def getRuntimeVersion() -> Tuple[Number, Number, Number]: ...
```
We can give a more precise type here, e.g., `int`, right?

ezyang commented on torch/backends/cudnn/__init__.py:
```diff
@@ -3,11 +3,12 @@
 import warnings
 from contextlib import contextmanager
 from torch.backends import ContextProp, PropModule, __allow_nonbracketed_mutation
+from torch.types import _bool
```
I don't think it should be necessary to use `_bool` here, as `torch.bool` isn't in scope here as `bool` (which is the reason for this workaround). Just use `bool` directly.

ezyang commented on torch/backends/cudnn/__init__.py:
```diff
@@ -82,13 +83,11 @@ def is_acceptable(tensor):
-
-_handles = {}
-verbose = False
+verbose = False  # might be unused
```
I agree. Can you file an issue about this and CC me?

ezyang reviewed (changes requested):
This looks good. Just some small nits and I'll land. Thanks for doing this!

ShawnZhong committed:
Fix `torch.backends.cudnn` mypy error

remove `mypy-torch.backends.cudnn.rnn`

Add type hints

minor

Fix flake8-py3

flake8

inline pyi

Update _cudnn.pyi

Update __init__.py
